File: 0020-valid-parentheses/solution.c
bool isValid(char* s) {
    char stack[10000];
    int top=-1;
    int n=strlen(s);
    for(int i=0;i<n;i++){
        if(s[i]=='{' || s[i]=='(' || s[i]=='['){
            top++;
            stack[top]=s[i];
        }else if(top==-1){
            return 0;
        }else if(s[i]=='}' || s[i]==')' ||s[i]==']'){
            if(s[i]=='}' && stack[top]=='{'){
                top--;
            }else if(s[i]==')' && stack[top]=='('){
                top--;
            }else if(s[i]==']' && stack[top]=='['){
                top--;
            }else{
                return 0;
            }
        }
    }
    if(top==-1){
        return 1;
    }
    return 0;
}

File: 0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(c=='{' || c=='[' || c=='('){
                stack.push(c);
            }
            else if(!stack.isEmpty()){
                if(c=='}' && stack.peek()!='{'){
                    return false;
                }else if(c==']' && stack.peek()!='['){
                    return false;
                }else if(c==')' && stack.peek()!='('){
                    return false;
                }
                stack.pop();
            }else{
                return false;
            }
        }
        return stack.isEmpty();
    }
}
